### PR TITLE
Extract rest invocation

### PR DESCRIPTION
**Pull request by wujimin**

### CODE REVIEW

File: common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+  protected void invoke() {
+    try {
+      this.initProduceProcessor();
+      this.setContext();
+      invocation.getHandlerContext().put(RestConst.REST_REQUEST, requestEx);
+
+      for (HttpServerFilter filter : httpServerFilters) {
+        Response response = filter.afterReceiveRequest(invocation, requestEx);
+        if (response != null) {
+          sendResponseQuietly(response);
```
Comment (WillemJiang) [inline common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java]: it looks like the filter could hijack the response message which calling the doInvoke method.
I don't think it's good idea, there are too much assumption.

File: common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestServer.java [hunk 2]
```diff
@@ -1,228 +0,0 @@
...
-import io.servicecomb.foundation.common.utils.JsonUtils;
-import io.servicecomb.foundation.common.utils.SPIServiceUtils;
-import io.servicecomb.foundation.vertx.http.HttpServletRequestEx;
-import io.servicecomb.foundation.vertx.http.HttpServletResponseEx;
-import io.servicecomb.foundation.vertx.stream.BufferOutputStream;
-import io.servicecomb.serviceregistry.RegistryUtils;
-import io.servicecomb.swagger.invocation.Response;
-import io.servicecomb.swagger.invocation.exception.InvocationException;
-
-public class AbstractRestServer {
```
Comment (WillemJiang) [inline common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestServer.java]: Is it safe to remove the AbstractRestServer?

File: common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java (hunk 1 above)
Comment (wujimin) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java]: modify response is one purpose of filter definition：
  producer generator signature and write it to response header.

File: common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestServer.java (hunk 2 above)
Comment (wujimin) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestServer.java]: this is our inner class, not a interface to developers

File: common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java (hunk 1 above)
Comment (WillemJiang) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java]: Please add some comments in the HttpServerFilter.afterReceiveRequst,  to make sure user know about this kind of feature.

Comment (wujimin) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java]: already have comments in HttpServerFilter.afterReceiveRequst and HttpClientFilter.afterReceiveResponse in current PR code.

image

image

Comment (WillemJiang) [reply common/common-rest/src/main/java/io/servicecomb/common/rest/AbstractRestInvocation.java]: OK.

**Pull request merged**